qt/python/mantidqt/mantidqt/widgets/sliceviewer/models/model.py

```diff
@@ -576,13 +596,27 @@ def _cut_names(self, cut: str):
 
         return xcut_name, ycut_name, help_msg
 
-    def get_hkl_from_xyz(self, xdim, ydim, zdim, xdata, ydata, zdata):
+    @staticmethod
+    def get_dim_indices(slice_point: List[float], qdims: List[int], transpose: bool) -> Tuple[int]:
+        """Returns the indices of the selected x and y axes. It also returns the index of the first non x or y axis."""
+        xdim, ydim = WorkspaceInfo.display_indices(slice_point, transpose)
+        # Get the index of the first axis which is not x or y. Here it is called z.
+        zdim = next(i for i, v in enumerate(slice_point) if v is not None and i in qdims)
+        return xdim, ydim, zdim
+
+    def get_hkl_from_full_point(self, full_point: List[float], qdims: List[int], xdim: int, ydim: int, zdim: int):
+        """Gets the values of h, k and l from a full point which can include 3 or more dimensions."""
         basis_transform = self.get_proj_matrix()
-        if basis_transform is not None:
-            hkl = basis_transform[:, xdim] * xdata + basis_transform[:, ydim] * ydata + basis_transform[:, zdim] * zdata
-        else:
-            hkl = {0.0, 0.0, 0.0}
-        return hkl
+        if basis_transform is None:
+            return 0.0, 0.0, 0.0
+
+        # Get the values for h, k and l
+        hkl_point = tuple(full_point[q_i] for q_i in qdims)
+
+        h_i, k_i, l_i = np.argsort([xdim, ydim, zdim])
```

Comment:
Again, `h_i` etc. correspond to the index of the `proj_matrix` column correpsonding to the index of the elements in `hkl_point` - perhaps `icol_x` etc. would be better names?